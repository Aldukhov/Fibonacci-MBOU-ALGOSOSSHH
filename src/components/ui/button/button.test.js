import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import { Button } from './button';


test('Btn with text', () => {
const { getByTestId } = render ( <Button text= 'Click'/>)
expect(getByTestId('btn')).toHaveTextContent('Click');
expect(getByTestId('btn')).toMatchSnapshot();
})

test('Btn without text', () => {
    const {getByTestId} = render (<Button />);

    expect(getByTestId('btn')).toBeInTheDocument();

    const pElement = getByTestId('btn').querySelector('p');

    expect(pElement).toBeInTheDocument();

    expect(pElement.textContent).toBe('');
    expect(getByTestId('btn')).toMatchSnapshot();
})

test('Btn disabled', () => {
    const {getByTestId} = render (<Button disabled/>);

    expect(getByTestId('btn')).toBeDisabled();
    expect(getByTestId('btn')).toMatchSnapshot();
})

test('Btn isLoader', () => {
    const {getByAltText} = render (<Button isLoader/>);

    expect(getByAltText('Загрузка.')).toBeInTheDocument();
    expect(getByAltText('Загрузка.')).toMatchSnapshot();     
})

test ('Btn onClick event', () => {
    const onClickMock = jest.fn();

    const { getByTestId } = render (<Button onClick={onClickMock}/>);

    fireEvent.click(getByTestId('btn'));

    expect(onClickMock).toHaveBeenCalledTimes(1);
    expect(getByTestId('btn')).toMatchSnapshot();
})