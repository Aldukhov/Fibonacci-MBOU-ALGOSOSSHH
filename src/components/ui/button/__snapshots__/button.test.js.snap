// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="btn"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Btn isLoader 1`] = `
<img
  alt="Загрузка."
  class="loader_icon"
  src="loader.svg"
/>
`;

exports[`Btn onClick event 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="btn"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Btn with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="btn"
  type="button"
>
  <p
    class="text undefined"
  >
    Click
  </p>
</button>
`;

exports[`Btn without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="btn"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
